### PR TITLE
Definition file absolute

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/1475)

### CODE REVIEW

**Comment by sschuberth** [comment]
Need to drop 980933a2833eec2a360b3b1cb901fc9112a47d5c again as the original code is required to wrap any exceptions.

File: analyzer/src/main/kotlin/PackageManager.kt [hunk 1]
```diff
@@ -187,14 +187,19 @@ abstract class PackageManager(
...
+     * for all [definitionFiles] which were found by searching the [analyzerRoot] directory. By convention, the
+     * [definitionFiles] must be absolute.
      */
     open fun resolveDependencies(definitionFiles: List<File>): ResolutionResult {
         val result = mutableMapOf<File, ProjectAnalyzerResult>()
 
         prepareResolution(definitionFiles)
 
         definitionFiles.forEach { definitionFile ->
+            require(definitionFile.isAbsolute) {
```
Comment (fviernau) [inline analyzer/src/main/kotlin/PackageManager.kt]: nit: Doing this in line 194, in particular before prepareResolution would be better in terms of 'fail early'.

Comment (fviernau) [reply analyzer/src/main/kotlin/PackageManager.kt]: Can you please also 'require' the definitionFile to be located underneath the analyzerRoot ?

Comment (sschuberth) [reply analyzer/src/main/kotlin/PackageManager.kt]: > 
> 
> nit: Doing this in line 194, in particular before prepareResolution would be better in terms of 'fail early'.

I didn't want to add another `forEach` loop, but yes, maybe you're right that the check should be done before calling `prepareResolution()`.

Comment (sschuberth) [reply analyzer/src/main/kotlin/PackageManager.kt]: > 
> 
> Can you please also 'require' the definitionFile to be located underneath the analyzerRoot ?

I can probably combine these, let's see...

File: analyzer/src/main/kotlin/PackageManager.kt [hunk 2]
```diff
@@ -187,11 +187,18 @@ abstract class PackageManager(
...
     /**
      * Return a tree of resolved dependencies (not necessarily declared dependencies, in case conflicts were resolved)
-     * for all [definitionFiles] which were found by searching the [analyzerRoot] directory.
+     * for all [definitionFiles] which were found by searching the [analyzerRoot] directory. By convention, the
+     * [definitionFiles] must be absolute.
      */
     open fun resolveDependencies(definitionFiles: List<File>): ResolutionResult {
         val result = mutableMapOf<File, ProjectAnalyzerResult>()
 
+        definitionFiles.forEach { definitionFile ->
```
Comment (fviernau) [inline analyzer/src/main/kotlin/PackageManager.kt]: I'd move it even before the declaration of result.

Comment (sschuberth) [reply analyzer/src/main/kotlin/PackageManager.kt]: Done.

**Pull request merged**